android/build_libraries_x86.sh

```diff
@@ -14,7 +14,7 @@ MAKECLEAN="yes"
 export BOINC=".." #BOINC source code
 
 export ANDROID_TC="${ANDROID_TC:-$HOME/android-tc}"
-export ANDROIDTC="${ANDROID_TC_X86:-$ANDROID_TC/x86"
+export ANDROIDTC="${ANDROID_TC_X86:-$ANDROID_TC/x86}"
 export TCBINARIES="$ANDROIDTC/bin"
```

Comment:
Why not treat the following variables like `ANDROIDTC` above? Yes, that's already fixed but you keep depending on that. Ideally you want to protect any variable, without further dependencies.

Reply:
Sorry, I don't understand what you are saying. Protect what variable, with which dependencies?

Reply:
While I always add curly braces (`{}`) around all variables in shell code *I* write, doing so is only required when doing any kind of variable substitution (ie: `${ANDROID_TC_X86:-Default}`). In practice you'll find lots and lots of shell code that only uses the braces when required and this won't cause any problems if you always consider wether the character *after* the variable name *may be part of a shell variable name* or not when writing it.

I tend to ofter *not* consider the above criterion in practice so I add the braces just to be safe, but this really only matters when you write the code; once it's properly written adding the braces everywhere is really not worth the (wasted) time, while also likely to cause hard to spot errors if the code is not reviewed very carefully (wasting everybody's time).